lib/options.js

```diff
@@ -12,23 +11,13 @@ export type OptionsFormat = {
 /**
  * The plugin options
  */
-class Options extends Record({
+const optionsMerger = (partialOptions: OptionsFormat) => ({
     maxDepth: 6,
     types: ['ul_list', 'ol_list'],
     typeItem: 'list_item',
     typeDefault: 'paragraph',
-    canMerge: (a: Node, b: Node) => a.type === b.type
-}) {
-    // Maximum level of nesting.
-    maxDepth: number;
-    // The possibles types for list containers
-    types: string[];
-    // The type of list items
-    typeItem: string;
-    // The type of default block in items
-    typeDefault: string;
-    // You can control here the automatic merging of adjacent lists
-    canMerge: (listA: Node, listB: Node) => boolean;
-}
+    canMerge: (a: Node, b: Node) => a.type === b.type,
+    ...partialOptions
+});
 
-export default Options;
+export default optionsMerger;
```

Comment:
I think we can later move this directly to the main file, it's not too complex to have it there now. Let's keep it like this for now, we can play around with it once we're at the finish line. 🏁